test/auth.ts

```diff
@@ -16,7 +16,7 @@ describe('Auth', function() {
     assert(isFinite(port), 'invalid test port')
 
     const app = new Koa()
-    const rpc = new JsonRpcAuth('https://testnet.steem.vc')
+    const rpc = new JsonRpcAuth('https://testnet.steem.vc', undefined, {addressPrefix: 'STX'})
```

Comment:
does this mean that passing tests has a dependency on the network and your testnet node being up?

Reply:
Yes it does. I agree depending on a 3rd-party node is not optimal, when the official testnet is launched my plan is to switch to that. That said I do take uptime of the testnet seriously.

Testing this in a meaningful way is very hard with mock data, for this implementation it could be done since we only depend on a `get_accounts` call but when switching back to `verify_signatures` it makes for a pretty useless test.

I think we always should test against a real steemd node, these tests help us not only discover bugs in this library but also in our entire ecosystem (for example the dsteem test suite has already found several bugs in steemd that has since been fixed). I've never understood why you would want to mock an api when testing against the real thing is a possibility (testnets, sandbox  versions, etc).

If we ship an update to steemd that breaks `verify_signatures` it should break this test, so should an update to jussi that mangles the request so the signature fails to verify.

```
watt deploy steemd latest to dev
watt test jsonrpc against dev
```

Reply:
> I've never understood why you would want to mock an api when testing against the real thing is a possibility

There are three reasons to write unit tests:
- _rapid_ exercise of code
- no dependency on external services for test to run
- enforcement of guarantees and invariants in codebase at time code was written

You have already described the very valid reasons for writing _integration_ tests.